src/ClientData/TracepointData.cpp

```diff
@@ -112,20 +104,20 @@ uint32_t TracepointData::GetNumTracepointEventsForThreadId(uint32_t thread_id) c
   return it->second.size();
 }
 
-bool TracepointData::AddUniqueTracepointInfo(uint64_t key,
-                                             orbit_grpc_protos::TracepointInfo tracepoint) {
+bool TracepointData::AddUniqueTracepointInfo(uint64_t key, TracepointInfo tracepoint) {
   absl::MutexLock lock{&unique_tracepoints_mutex_};
-  auto [unused_it, inserted] = unique_tracepoints_.try_emplace(key, std::move(tracepoint));
+  auto [unused_it, inserted] =
+      unique_tracepoints_.try_emplace(key, std::make_unique<TracepointInfo>(std::move(tracepoint)));
   return inserted;
 }
 
-orbit_grpc_protos::TracepointInfo TracepointData::GetTracepointInfo(uint64_t hash) const {
+const TracepointInfo* TracepointData::GetTracepointInfo(uint64_t hash) const {
   absl::MutexLock lock{&unique_tracepoints_mutex_};
   auto it = unique_tracepoints_.find(hash);
   if (it != unique_tracepoints_.end()) {
-    return it->second;
+    return it->second.get();
```

Comment:
So do I get it correctly: changing to returning a pointer that is going to be dereferenced without holding the mutex is fine because `unique_tracepoints_` has been changed to hold `unique_ptr`s? I would add a comment on `unique_tracepoints_` then.

Reply:
Exactly. I added comment.